contracts/refs/OracleRefV2.sol

```diff
@@ -80,6 +80,23 @@ abstract contract OracleRefV2 is IOracleRefV2, CoreRefV2 {
         return value;
     }
 
+    /// @notice returns inverse of normal value. Used in NonCustodial PSM
+    /// to normalize decimals to properly deplete the buffer in the
+    /// Global System Exit Rate Limiter
+    function getExitValue(uint256 amount) public view returns (uint256) {
+        uint256 scalingFactor;
+        if (decimalsNormalizer == 0) {
+            return amount;
+        }
+        if (decimalsNormalizer < 0) {
+            scalingFactor = 10 ** uint256(-decimalsNormalizer);
+            return amount * scalingFactor;
+        } else {
+            scalingFactor = 10 ** uint256(decimalsNormalizer);
+            return amount / scalingFactor;
+        }
+    }
+
```

Comment:
I'm not very fond of adding this very "exit-specific" function at the generic OracleRef level. Maybe you could have a function `one()` that returns `1e18 * scalingFactor` -> it will be 1e30 for USDC's oracleref because the decimal correction is 12. Then in the NCPSM, you deplete/replenish by amountUSDC * one() / 1e18.

Reply:
or just move the function to the NCPSM directly

Reply:
also if you keep the function, memoize `decimalsNormalizer` to avoid loading state 3 times

Reply:
ok, will move to the NCPSM and memoize the decimalsNormalizer

Reply:
decimalsNormalizer is immutable so it's unlikely we're saving gas by pushing to the stack